tests/unit/job/test_query.py

```diff
@@ -1682,6 +1682,75 @@ def test_result_with_start_index(self):
             tabledata_list_request[1]["query_params"]["maxResults"], page_size
         )
 
+    def test_result_with_start_index_multi_page(self):
+        # When there are multiple pages of response and the user has set
+        # start_index, we should supply start_index to the server in the first
+        # request. However, in the subsequent requests, we will pass only
+        # page_token but not start_index, because the server only allows one
+        # of them.
+        from google.cloud.bigquery.table import RowIterator
+
+        query_resource = {
+            "jobComplete": True,
+            "jobReference": {"projectId": self.PROJECT, "jobId": self.JOB_ID},
+            "schema": {"fields": [{"name": "col1", "type": "STRING"}]},
+            "totalRows": "7",
+        }
+        tabledata_resource_1 = {
+            "totalRows": "7",
+            "pageToken": "page_token_1",
+            "rows": [
+                {"f": [{"v": "abc"}]},
+                {"f": [{"v": "def"}]},
+                {"f": [{"v": "ghi"}]},
+            ],
+        }
+        tabledata_resource_2 = {
+            "totalRows": "7",
+            "pageToken": None,
+            "rows": [
+                {"f": [{"v": "jkl"}]},
+                {"f": [{"v": "mno"}]},
+                {"f": [{"v": "pqe"}]},
+            ],
+        }
+
```

Comment:
Please go through the rest of this test and provide some additional comments.

Clarify why we are using `total_rows` of `7`.
Clarify why setting `total_rows` to `7` despite only have six rows of tabledata is ok/reasonable
Clarify what is at the indexed positions:

* `tabledata_list_request_1[1]`
* `tabledata_list_request_2[1]`

Reply:
> Clarify why we are using total_rows of 7.
Clarify why setting total_rows to 7 despite only have six rows of tabledata is ok/reasonable

Done.

> Clarify what is at the indexed positions:
tabledata_list_request_1[1]
tabledata_list_request_2[1]

There are comments above the code block explaining that these are the first and the second requests that the client sends to the server.

Reply:
I am gonna drop this comment.